src/DataModel/IMetadataProvider.ts

```diff
@@ -0,0 +1,30 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import "whatwg-fetch";
+import {IData} from "./IData";
+
+/**
+ * @interface Describes an abstract meta-data providing facility which translates from raw @link Response to an abstract data model.
+ */
+export interface IMetadataProvider
```

Comment:
Could you please elaborate a bit on why this class is needed? What is the abstract data model? IData and IMetadata? Something else?